app/root.tsx

```diff
@@ -25,6 +28,55 @@ import iconsHref from "~/icons.svg";
 import cx from "clsx";
 import { canUseDOM } from "./ui/primitives/utils";
 import { GlobalLoading } from "./ui/global-loading";
+import { type Route } from "./+types/root";
+
+const redirectOldDocs: Route.unstable_MiddlewareFunction = ({ request }) => {
+  const { pathname } = new URL(request.url);
+
+  if (pathname === "/docs") {
+    throw redirect(`https://v2.remix.run/docs`);
+  }
+
+  if (!pathname.startsWith("/docs/")) {
+    return;
+  }
```

Comment:
Added this back now that we're not automatically constrained to `/docs` URLs based on the route file matching